doc/examples/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,303 @@
+    robot_goal_state->setJointPositions("panda_joint5", &panda_joint5);
+    robot_goal_state->setJointPositions("panda_joint6", &panda_joint6);
+    robot_goal_state->setJointPositions("panda_joint7", &panda_joint7);
+    robot_goal_state->setJointPositions("panda_joint3", &panda_joint3);
+
+    // Set goal state
+    planning_component_->setGoal(*robot_goal_state);
+  }
+
+  /// \brief Set goal state for next planning attempt based on query loaded from the database
+  void setQueryGoal()
+  {
+    // Set goal state
+    if (!planning_query_request_.goal_constraints.empty())
+    {
+      planning_component_->setGoal(planning_query_request_.goal_constraints);
+    }
+    else
+      RCLCPP_ERROR(LOGGER, "Planning query request does not contain any goal constraints");
+  }
+
+  /// \brief Request a motion plan based on the assumption that a goal is set and print debug information.
+  void planAndPrint()
+  {
+    // Set start state as current state
+    planning_component_->setStartStateToCurrentState();
+
+    moveit_cpp::PlanningComponent::MultiPipelinePlanRequestParameters multi_pipeline_plan_request{
+      node_, { "ompl_rrtc", "pilz_lin", "chomp" }
+    };
+
+    auto plan_solution = planning_component_->plan(multi_pipeline_plan_request, &getShortestSolution);
+
+    // Check if PlanningComponents succeeded in finding the plan
+    if (plan_solution)
+    {
+      // Visualize the trajectory in Rviz
+      auto robot_model_ptr = moveit_cpp_->getRobotModel();
+      auto robot_start_state = planning_component_->getStartState();
+      auto joint_model_group_ptr = robot_model_ptr->getJointModelGroup(PLANNING_GROUP);
+
+      visual_tools_.publishTrajectoryLine(plan_solution.trajectory_, joint_model_group_ptr);
+      visual_tools_.trigger();
+    }
+
+    planning_component_->execute();
+
+    // Start the next plan
+    visual_tools_.prompt("Press 'next' in the RvizVisualToolsGui window to continue the demo");
+    visual_tools_.deleteAllMarkers();
+    visual_tools_.trigger();
+  }
+
+private:
+  std::shared_ptr<rclcpp::Node> node_;
+  std::shared_ptr<moveit_cpp::MoveItCpp> moveit_cpp_;
+  std::shared_ptr<moveit_cpp::PlanningComponent> planning_component_;
+  moveit_visual_tools::MoveItVisualTools visual_tools_;
+  moveit_msgs::msg::MotionPlanRequest planning_query_request_;
+};
+}  // namespace parallel_planning_example
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+  rclcpp::NodeOptions node_options;
+  RCLCPP_INFO(LOGGER, "Initialize node");
+
+  node_options.automatically_declare_parameters_from_overrides(true);
+  rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("parallel_planning_example", "", node_options);
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+  executor.add_node(node);
+  std::thread([&executor]() { executor.spin(); }).detach();
+
+  parallel_planning_example::Demo demo(node);
+
+  /* Otherwise robot with zeros joint_states */
```

Comment:
Is this comment correct here?

Reply:
It works without the wait :+1: although I suspect that this is mainly because of the wait for user input a few lines down the code. But we don't need it here